### PR TITLE
Add gradle project

### PR DESCRIPTION
**Pull request by lamtev**

### CODE REVIEW

File: api/src/test/java/org/accula/api/routers/GreetingRouterTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+@RunWith(SpringRunner.class)
+public final class GreetingRouterTest {
+    private static final String GREETING = "ACCULA is greeting you, Alice";
+    private static final String ERROR = "Missing required query param \"name\"";
+
+    @Autowired
+    private RouterFunction<ServerResponse> greetingRoute;
+
+    @Test
+    public void testGreetingRouteOk() {
```
Comment (vaddya) [inline api/src/test/java/org/accula/api/routers/GreetingRouterTest.java]: It's a matter of taste, but
```
        WebTestClient
                .bindToRouterFunction(greetingRoute)
                .build()
                .get().uri("/greet?name=Alice").exchange()
                .expectStatus().isOk()
                .expectBody(String.class).isEqualTo(GREETING);
```
seems to be more concise without loss of readability.

File: api/src/main/java/org/accula/api/config/HandlersConfig.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import org.accula.api.handlers.GreetingHandler;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootConfiguration
+public class HandlersConfig {
+    @Bean
+    @NotNull
+    public GreetingHandler greetingHandler() {
```
[the remark this answers is not in the document]
Comment (lamtev) [reply api/src/main/java/org/accula/api/config/HandlersConfig.java]: Completely agree with you

File: api/src/main/java/org/accula/api/routers/GreetingRouter.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.web.reactive.function.server.RouterFunctions;
+import org.springframework.web.reactive.function.server.ServerResponse;
+
+@Component
+@RequiredArgsConstructor
+public final class GreetingRouter {
+    private final GreetingHandler handler;
+
+    @Bean
+    @NotNull RouterFunction<ServerResponse> greetingRoute() {
```
[the remark this answers is not in the document]
Comment (lamtev) [reply api/src/main/java/org/accula/api/routers/GreetingRouter.java]: And `public` too I suppose

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/.idea
```
[the remark this answers is not in the document]
Comment (lamtev) [reply .gitignore]: I'll do the changes

File: api/src/main/java/org/accula/api/config/HandlersConfig.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (lamtev) [reply api/src/main/java/org/accula/api/config/HandlersConfig.java]: done!

File: api/src/main/java/org/accula/api/routers/GreetingRouter.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (lamtev) [reply api/src/main/java/org/accula/api/routers/GreetingRouter.java]: done!

File: .gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (lamtev) [reply .gitignore]: done!

File: api/src/test/java/org/accula/api/routers/GreetingRouterTest.java (hunk 1 above)
Comment (lamtev) [reply api/src/test/java/org/accula/api/routers/GreetingRouterTest.java]: done!

**Pull request merged**